src/Functions/FunctionsStringDistance.cpp

```diff
@@ -0,0 +1,218 @@
+        const ColumnString::Chars & data,
+        const ColumnString::Offsets & offsets,
+        const std::string & needle,
+        PaddedPODArray<ResultType> & res)
+    {
+        constantVector(needle, data, offsets, res);
+    }
+
+};
+
+struct ByteHammingDistanceImpl
+{
+    using ResultType = UInt64;
+    static ResultType inline process(
+        const char * __restrict haystack, size_t haystack_size, const char * __restrict needle, size_t needle_size)
+    {
+        UInt64 res = 0;
+        const char * haystack_end = haystack + haystack_size;
+        const char * needle_end = needle + needle_size;
+
+#ifdef __SSE4_2__
+        static constexpr auto mode = _SIDD_UBYTE_OPS | _SIDD_CMP_EQUAL_EACH | _SIDD_NEGATIVE_POLARITY;
+
+        const char * haystack_end16 = haystack + haystack_size / 16 * 16;
+        const char * needle_end16 = needle + needle_size / 16 * 16;
+
+        for (; haystack < haystack_end16 && needle < needle_end16; haystack += 16, needle += 16)
+        {
+            __m128i s1 = _mm_loadu_si128(reinterpret_cast<const __m128i *>(haystack));
+            __m128i s2 = _mm_loadu_si128(reinterpret_cast<const __m128i *>(needle));
+            auto result_mask = _mm_cmpestrm(s1, 16, s2, 16, mode);
+            const __m128i mask_hi = _mm_unpackhi_epi64(result_mask, result_mask);
+            res += _mm_popcnt_u64(_mm_cvtsi128_si64(result_mask)) + _mm_popcnt_u64(_mm_cvtsi128_si64(mask_hi));
+        }
+#endif
+        for (; haystack != haystack_end && needle != needle_end; ++haystack, ++needle)
+            res += *haystack != *needle;
+
+        res = res + (haystack_end - haystack) + (needle_end - needle);
+        return res;
+    }
+};
+
+struct ByteJaccardIndexImpl
+{
+    using ResultType = Float64;
+    static ResultType inline process(
+        const char * __restrict haystack, size_t haystack_size, const char * __restrict needle, size_t needle_size)
+    {
+        if (haystack_size == 0 || needle_size == 0)
+            return 0;
+
+        std::unordered_set<char> haystack_set(haystack, haystack + haystack_size);
+        std::unordered_set<char> needle_set(needle, needle + needle_size);
+
+        size_t intersect = 0;
+        for (auto elem : haystack_set)
+        {
+            intersect += needle_set.contains(elem);
+        }
+        return static_cast<Float64>(intersect) / (haystack_set.size() + needle_set.size() - intersect);
+    }
+};
+
+struct ByteEditDistanceImpl
+{
+    using ResultType = UInt64;
+    static constexpr size_t max_string_size = 1u << 16;
+
+    static ResultType inline process(
+        const char * __restrict haystack, size_t haystack_size, const char * __restrict needle, size_t needle_size)
+    {
+        if (haystack_size == 0 || needle_size == 0)
+            return haystack_size + needle_size;
+
+        /// Safety threshold against DoS, since we use two array to calculate the distance.
+        if (haystack_size > max_string_size || needle_size > max_string_size)
```

Comment:
The check in `FunctionsStringSimilarity.h` is used for functions in `FunctionsStringSimilarity.cpp`, like `ngramDistance`, because `FunctionDistanceImpl` does not have `max_string_size` element:
```cpp
if constexpr (has_max_string_size<Impl>)
{
   ...
} 
```